index.js

```diff
@@ -95,6 +95,39 @@ class Conf {
 		this.store = plainObject();
 	}
 
+	static isEqual(newValue, oldValue) {
+		try {
+			// TODO: Use `util.isDeepStrictEqual` when targeting Node.js 10
+			assert.deepEqual(newValue, oldValue);
+		} catch (_) {
+			return false;
+		}
+
+		return true;
+	}
+
+	onDidAnyChange(callback) {
+		if (typeof callback !== 'function') {
+			throw new TypeError(`Expected \`callback\` to be of type \`function\`, got ${typeof callback}`);
+		}
+
+		let currentValue = this.store;
+
+		const onChange = () => {
+			const oldValue = currentValue;
+			const newValue = this.store;
+
+			const didChange = Conf.isEqual(oldValue, newValue) === false;
+			if (didChange) {
+				currentValue = newValue;
+				callback.call(this, newValue, oldValue);
+			}
+		};
```

Comment:
`onChange` repeats some logic as in `onDidChange`, maybe it could be generalized.

Reply:
[dot-prop](https://github.com/sindresorhus/dot-prop) supports [returning the entire object](https://github.com/sindresorhus/dot-prop/blob/42cc3bf8ec8a404bca121a70ea831a9d687582a2/index.js#L24-L26) when a path is excluded:

```js
	get(obj, path, value) {
		if (!isObj(obj) || typeof path !== 'string') {
			return value === undefined ? obj : value;
		}
```

This should pave the path to greatly simplify the two implementations since `onDidAnyChange` will just be `onDidChange` with an empty `key`. To support separate validation, the internals could be extracted into a shared method.